### ISSUE TITLE
Corrupt Open XML: The specified package is invalid. The main part is missing.

### ISSUE THREAD

**Issue by GuacoIV** [EWSoftware/SHFB#915]
When I follow [this walkthrough](http://ewsoftware.github.io/SHFB/html/d5dbf017-25f1-48cb-86db-90f76ee8f4d7.htm) (or make my own Sandcastle Help File Builder Project against my own projects) and change the "Presentation style" over to "Open XML Document" the resultant .docx output always gives the following errors when opening in Microsoft Word:
![image](https://user-images.githubusercontent.com/6468346/172453887-fbbd7e94-270e-4064-8ca5-095d3a3c9238.png)
![image](https://user-images.githubusercontent.com/6468346/172453971-9411ab97-c774-4e35-95d0-028474b2c4e2.png)

Attempting to follow the troubleshooting steps [here](https://tunnelvisionlabs.github.io/SHFB/docs-master/SandcastleBuilder/html/c5e7e920-2cb4-4f78-b167-634361d85263.htm#Troubleshooting), I hit an obstacle in that there is never a line number given in the error messages in Word. I decided to use the [OpenXmlValidator](https://docs.microsoft.com/en-us/dotnet/api/documentformat.openxml.validation.openxmlvalidator?view=openxml-2.8.1) class with the following code in Linqpad:

`OpenXmlValidator blah = new OpenXmlValidator();
blah.Validate(WordprocessingDocument.Open(@"C:\dev\APIDoc\Help\APIDoc.docx", false)).Dump();`

but it fails in the Open method before even getting to the validator with the following exception:
![image](https://user-images.githubusercontent.com/6468346/172455069-50bf6629-4da8-4ff7-bd9c-e315077b1c65.png)

**Comment by EWSoftware** [comment]
There's currently a problem with how the document file is being created.  It used to work but fails now due to what appears to be a change in how the .NET framework code handles the folder names within the package file.

I'm currently reworking how the presentation styles work and I'm working through the Open XML presentation style now and have fixed that issue.  It will be a few weeks before it is ready though.  If you need support for them now, you might try [version 2021.4.9.0](https://github.com/EWSoftware/SHFB/releases/tag/v2021.4.9.0) as it uses a different method of producing the document file.

**Comment by GuacoIV** [comment]
I see. Thanks for the information and workaround suggestion. Would be cool to see what the change ends up being though I suppose the pull request will have a lot of other content based on what you're saying. Feel free to set the status of this as you wish or keep open to mark that its not released yet. Thanks!

**Comment by EWSoftware** [comment]
It was a minor update.  It was working when I first updated the code last year to support dotnet build by switching to the built-in packaging and compression classes.   At some point after that, the package code in the framework appears to have been updated to conform to the spec such that backslashes in paths are not valid and should be forward slashes and paths cannot start with a leading path separator.  WinZip and PKZIP could still read and extract the files and didn't care about the format of the paths.  When I repacked the document files using either of them I noticed the change in the paths in the zip file directory which gave me something to look for and helped me find the reference to the change and the fix.
